core/src/main/scala/cats/Show.scala

```diff
@@ -42,18 +42,12 @@ object Show extends ScalaVersionSpecificShowInstances with ShowInstances {
   /**
    * creates an instance of [[Show]] using the provided function
    */
-  def show[A](f: A => String): Show[A] =
-    new Show[A] {
-      def show(a: A): String = f(a)
-    }
+  def show[A](f: A => String): Show[A] = f(_)
```

Comment:
Is there a practical difference between these? e.g. no class is emitted.

Reply:
Yes, indeed 👍 - because `Show` has only one method

Reply:
Right, thanks. Is this affected by the aforementioned Scala 2 bug?

Also, can `Semigroup` etc. get the same treatment?
https://github.com/typelevel/cats/blob/6ba7e4b782c0355c7fa2b5a787355a19b22d469a/kernel/src/main/scala/cats/kernel/Semigroup.scala#L151-L154

Reply:
I think for `Semigroup` it doesn't matter because it has other (non-abstract) methods. So then the bug applies and there would be an anonymous classes generated even if we use the SAM syntax. So the only benefit would be aesthetics.

Reply:
I see, so it's not just that `Show` has one abstract method, it's that it has no other methods.

I agree it's only aesthetics on Scala 2, but does the bug apply to Scala 3 as well?

Reply:
> I agree it's only aesthetics on Scala 2, but does the bug apply to Scala 3 as well?

I don't remember - that's a good question.

Reply:
The reason I ask is, I don't care so much about the handful of `Semigroup.instance` etc. that can be re-written.

But I'm wondering if in Scala 3 the boilerplate instances themselves could be written directly like this. So we get the win in terms of jar size without introducing any indirection at all.

Reply:
Even if that's possible it would be quite hard to split like this 😄

Reply:
FTR Scala 3 doesn't have this bug - but I again I'm sceptical about version-specific boilerplate generators 🤔

Reply:
If there's interest it can be a followup PR. I think it shouldn't be _too_ hard (famous last words): currently all the instances are in the same file I think, when they could easily be split among a few files. And some of those files can go into the scala-2 and scala-3 srcs.